### PR TITLE
LabeledZippedImageDataset

### PR DESCRIPTION
**Pull request by d0i**
As there was request to add label handling for ZippedImageDataset, I wrote this.
This only take label file and do not accept tuples, unlike LabeledImageDataset.

### CODE REVIEW

File: chainer/datasets/image_dataset.py [hunk 1]
```diff
@@ -163,6 +163,53 @@ def get_example(self, i):
         return _postprocess_image(image), label
 
 
+class LabeledZippedImageDataset(dataset_mixin.DatasetMixin):
+    """Dataset of zipped image and label pairs built from a list of paths
+    and labels.
+
+    This dataset is zip version of `LabeledImageDataset`. It takes a
```
Comment (kmaehashi) [inline chainer/datasets/image_dataset.py]: Please make it cross-link:
```
:class:`LabeledImageDataset`
```

File: chainer/datasets/image_dataset.py [hunk 2]
```diff
@@ -163,6 +163,53 @@ def get_example(self, i):
...
+    zipfile like `ZippedImageDataset`. The label file shall contain
```
Comment (kmaehashi) [inline chainer/datasets/image_dataset.py]: ditto

File: chainer/datasets/image_dataset.py [hunk 3]
```diff
@@ -163,6 +163,53 @@ def get_example(self, i):
...
+    lines like text file used in :class:`LabeledImageDataset`, but a
+    filename in each line of the label file shall match with a file in
+    the zip archive.
+
+    Args:
+        zipfilename (str): Path to a zipfile with images
+        labelfilename (str): Path to a label file. ``i``-th line shall
+            contain a filename and an integer label and it corresponds
```
Comment (kmaehashi) [inline chainer/datasets/image_dataset.py]: I think `s/and it/that/` improves readability.

File: chainer/datasets/image_dataset.py [hunk 4]
```diff
@@ -163,6 +163,52 @@ def get_example(self, i):
...
+                 label_dtype=numpy.int32):
+        _check_pillow_availability()
+        pairs = []
+        with open(labelfilename) as pairs_file:
+            for i, line in enumerate(pairs_file):
+                pair = line.strip().split()
+                if len(pair) != 2:
+                    raise ValueError(
+                        'invalid format at line {} in file {}'.format(
+                            i, pairs_path))
```
Comment (kmaehashi) [inline chainer/datasets/image_dataset.py]: `s/pairs_path/pairs_file/`

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
Please check comments and Travis failures.

**Comment by d0i** [comment]
@kmaehashi Thanks for the comments!

**Comment by kmaehashi** [comment]
Thanks!

Jenkins, test this please.

**Comment by kmaehashi** [comment]
Ah, could you check Travis failure? (just a syntax issue)

```
$ autopep8 -r . --diff | tee check_autopep8
--- original/./chainer/datasets/image_dataset.py
+++ fixed/./chainer/datasets/image_dataset.py
@@ -185,6 +185,7 @@
         label_dtype: Data type of the labels.
 
     """
+
     def __init__(self, zipfilename, labelfilename, dtype=None,
                  label_dtype=numpy.int32):
         _check_pillow_availability()
```

**Comment by d0i** [comment]
Sorry for an easy mistake!

**Comment by kmaehashi** [comment]
Thanks for update! Jenkins, test this please.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**